File: src/experiments/monolith/WORKSPACE.cc

######################################
# Language: C++
######################################
# - releases: https://github.com/bazelbuild/rules_cc/releases
# - questions
#    - how do I specify g++ instead of clang?

# https://bazel.build/docs/cc-toolchain-config-reference
#   GCC
#    http_archive(
#        name = "aspect_gcc_toolchain",
#        sha256 = "3341394b1376fb96a87ac3ca01c582f7f18e7dc5e16e8cf40880a31dd7ac0e1e",
#        strip_prefix = "gcc-toolchain-0.4.2",
#        urls = [
#            "https://github.com/aspect-build/gcc-toolchain/archive/refs/tags/0.4.2.tar.gz",
#        ],
#    )
#
#    load("@aspect_gcc_toolchain//toolchain:repositories.bzl", "gcc_toolchain_dependencies")
#    gcc_toolchain_dependencies()
#    load("@aspect_gcc_toolchain//toolchain:defs.bzl", "gcc_register_toolchain", "ARCHS")
#    gcc_register_toolchain(
#        name = "gcc_toolchain_x86_64",
#        target_arch = ARCHS.x86_64,
#    )

# Unit Testing in C++
#  - https://google.github.io/googletest/quickstart-bazel.html
#  - https://google.github.io/googletest/primer.html
#  - TODO(wmh): How do I establish the sha256 value?
http_archive(
    name = "gtest",
    url = "https://github.com/google/googletest/archive/release-1.12.0.zip",
    #sha256 = "b58cb7547a28b2c718d1e38aee18a3659c9e3ff52440297e965f5edffe34b6d0",
    #build_file = "gtest.BUILD",
    strip_prefix = "googletest-release-1.12.0",
)

File: src/experiments/monolith/WORKSPACE.py

######################################
# Language: Python
######################################
# - releases: https://github.com/bazelbuild/rules_python/releases
# - tutorials
#    - https://earthly.dev/blog/build-and-deploy-pyapp-with-bazel/
# - python parsers
#    - https://docs.python.org/3.9/library/parser.html
#    - https://github.com/lark-parser/lark

# WORKSPACE: https://github.com/bazelbuild/rules_python/releases "Paste this snippet into your WORKSPACE file:"

http_archive(
    name = "rules_python",
    sha256 = "e85ae30de33625a63eca7fc40a94fea845e641888e52f32b6beea91e8b1b2793",
    # sha256 = "9acc0944c94adb23fba1c9988b48768b1bacc6583b52a2586895c5b7491e2e31",
    strip_prefix = "rules_python-0.27.1",
    url = "https://github.com/bazelbuild/rules_python/releases/download/0.27.1/rules_python-0.27.1.tar.gz",
    # url = "https://github.com/bazelbuild/rules_python/releases/download/0.27.0/rules_python-0.27.0.tar.gz",
)

load("@rules_python//python:repositories.bzl", "py_repositories")
py_repositories()
